### DISCUSSION TITLE
How to check for OS without wxwidgets macros?

### DISCUSSION THREAD

**Discussion by falkTX** [audacity/audacity#882] [Q&A] [answered]
Currently Audacity codebase is using wxwidgets macros like `__WXMSW__` to check the target OS to build for.
This breaks a potential Qt toolkit build, because `__WXMSW__` is not defined on Windows though we can still use Windows code just fine, or macOS, etc.

What is best to use here? native macros like `_WIN32` and `__APPLE__`?
Welcome to suggestions.

Deciding on this is a blocker for pushing experimental Qt support.

**Comment by Be-ing** [comment, marked as answer]
One approach is to use CMake's OS detection and set `#define`s conditionally based on that: https://github.com/mixxxdj/mixxx/blob/main/CMakeLists.txt#L1093

**Reply by JamesCrook** [reply]
My thoughts too.

Another option is to put wxWidgets macros into some ubiquitously included .h file of our own.  Audacity.h or MemoryX.h.  Either just as a temporary measure so as to have to change less code.

Then at some later point we do a search and replace.
